cmake/alpakaCommon.cmake

```diff
@@ -506,31 +507,38 @@ endif()
 # Find HIP.
 if(alpaka_ACC_GPU_HIP_ENABLE)
 
-    # supported HIP version range
-    set(_alpaka_HIP_MIN_VER 5.1)
-    set(_alpaka_HIP_MAX_VER 5.7)
-    set(_alpaka_HIP_NEXT_MAJOR_MIN_VER 6.0)
+    check_language(HIP)
+
+    if(CMAKE_HIP_COMPILER)
+        enable_language(HIP)
+        find_package(hip REQUIRED)
 
-    find_package(hip "${_alpaka_HIP_MIN_VER}...${_alpaka_HIP_MAX_VER}")
+        set(_alpaka_HIP_MIN_VER 5.1)
+        set(_alpaka_HIP_MAX_VER 6.0)
 
-    if(NOT TARGET hip)
-        message(WARNING "Could not find HIP <=v${_alpaka_HIP_MAX_VER}. Now searching for unsupported HIP >v${_alpaka_HIP_MAX_VER}")
-        find_package(hip "${_alpaka_HIP_MAX_VER}")
-        if(NOT TARGET hip)
-            message(WARNING "Could not find HIP >=v${_alpaka_HIP_MAX_VER}. Now searching for unsupported HIP >v${_alpaka_HIP_NEXT_MAJOR_MIN_VER}")
-            find_package(hip "${_alpaka_HIP_NEXT_MAJOR_MIN_VER}")
+        # construct hip version only with major and minor level
+        # cannot use hip_VERSION because of the patch level
+        # 6.0 is smaller than 6.0.1234, so _alpaka_HIP_MAX_VER would have to be defined with a large patch level or
+        # the next minor level, e.g. 6.1, would have to be used.
+        set(_hip_MAJOR_MINOR_VERSION "${hip_VERSION_MAJOR}.${hip_VERSION_MINOR}")
+
+        if(${_hip_MAJOR_MINOR_VERSION} VERSION_LESS ${_alpaka_HIP_MIN_VER} OR ${_hip_MAJOR_MINOR_VERSION} VERSION_GREATER ${_alpaka_HIP_MAX_VER})
+            message(WARNING "HIP ${_hip_MAJOR_MINOR_VERSION} is not official supported by alpaka. Supported versions: ${_alpaka_HIP_MIN_VER} - ${_alpaka_HIP_MAX_VER}")
         endif()
-    endif()
 
-    if(NOT TARGET hip)
-        message(FATAL_ERROR "Optional alpaka dependency HIP could not be found!")
-    else()
-        target_link_libraries(alpaka INTERFACE hip::host hip::device)
+
+        target_link_libraries(alpaka INTERFACE "$<$<LINK_LANGUAGE:CXX>:hip::host>")
+        alpaka_set_compiler_options(HOST_DEVICE target alpaka "$<$<COMPILE_LANGUAGE:CXX>:-D__HIP_PLATFORM_AMD__>")
+        # only required for HIP 5.1 and maybe older
+        alpaka_set_compiler_options(HOST_DEVICE target alpaka "$<$<COMPILE_LANGUAGE:CXX>:-D__HIP_PLATFORM_HCC__>")
```

Comment:
If this is only required for HIP 5.1 we should apply the fix only for this version.

```
if(${_hip_MAJOR_MINOR_VERSION} VERSION_EQUAL "5.1")
        alpaka_set_compiler_options(HOST_DEVICE target alpaka "$<$<COMPILE_LANGUAGE:CXX>:-D__HIP_PLATFORM_HCC__>")
endif()
```